js/noms/src/database.js

```diff
@@ -89,7 +93,8 @@ export default class Database {
       const currentHeadRef = await currentDatasets.get(datasetId);
       if (currentHeadRef) {
         if (equals(commitRef, currentHeadRef)) {
-          return this;
+          // $FlowIssue: thinks commitRef is a Promise
+          return commitRef;
```

Comment:
I spent an hour trying to figure out what I'm doing wrong, here. Without these, flow complains that commitRef is a Promise, and that's incompatible with Ref. But...commitRef is the return value of `this.writeValue()`, which returns...a Ref! Can you help me figure out what the problem is?